kit/ForKit.cpp

```diff
@@ -327,23 +342,28 @@ static void cleanupChildren()
                                           << childJails.size() << " left: " << oss.str());
     }
 
-    if (segFaultCount)
-    {
-        if (WSHandler)
+    auto sendCountMessage = [](const std::string &countName, int count){
+        if (count)
         {
-            std::stringstream stream;
-            stream << "segfaultcount " << segFaultCount << '\n';
-            int ret = WSHandler->sendMessage(stream.str());
-            if (ret == -1)
+            if (WSHandler)
             {
-                LOG_WRN("Could not send 'segfaultcount' message through websocket");
-            }
-            else
-            {
-                LOG_WRN("Successfully sent 'segfaultcount' message " << stream.str());
+                std::stringstream stream;
+                stream << countName << count << '\n';
+                int ret = WSHandler->sendMessage(stream.str());
+                if (ret == -1)
+                {
+                    LOG_WRN("Could not send '" << countName << "' message through websocket");
+                }
+                else
+                {
+                    LOG_WRN("Successfully sent '" << countName << "' message " << stream.str());
+                }
             }
         }
-    }
+    };
+
+    sendCountMessage( "segfaultcount", segFaultCount);
```

Comment:
One message is better than three - I'd pack all three numbers into one message =)  Of course - we would expect some code on the other side in COOLWSD.cpp(?) to read & interpret that - and this would save duplication there too - you'll get to that bit too (?)

Anyhow - code flow looking better :-)

Reply:
Handled now.